packages/alchemy/src/react/context.ts

```diff
@@ -0,0 +1,46 @@
+"use client";
+
+import type { QueryClient } from "@tanstack/react-query";
+import { createContext, createElement, useContext } from "react";
+import type { AlchemyAccountsConfig } from "../config";
+import { NoAlchemyAccountContextError } from "./errors.js";
+
+export type AlchemyAccountContextProps =
+  | {
+      config: AlchemyAccountsConfig;
+      queryClient: QueryClient;
+    }
+  | undefined;
+
+export const AlchemyAccountContext = createContext<
+  AlchemyAccountContextProps | undefined
+>(undefined);
+
+export type AlchemyAccountsProviderProps = {
+  config: AlchemyAccountsConfig;
+  queryClient: QueryClient;
+};
+
+export const useAlchemyAccountContext = () => {
+  const context = useContext(AlchemyAccountContext);
+
+  if (context === undefined) {
+    throw new NoAlchemyAccountContextError("useAlchemyAccountContext");
+  }
+
+  return context;
+};
+
+export const AlchemyAccountProvider = ({
+  config,
+  queryClient,
+  children,
+}: React.PropsWithChildren<AlchemyAccountsProviderProps>) => {
+  // Note: we don't use .tsx because we don't wanna use rollup or similar to bundle this package.
+  // This lets us continue to use TSC for building the packages which preserves the "use client" above
```

Comment:
to expand a bit more, when I first prototyped this outside of this repo, I was initially using rollup to build. This was fine except that it wouldn't preserve the `use client` directive so some hooks weren't working well in nextjs. 

We'll need to figure this out when we add UI components, but for hooks I think this is fine for now

Reply:
the hooks files don't actually have any `jsx` syntax in them so we don't have to do that there